### PR TITLE
feat(forms): Add `parse` and `format` for FormikCheckboxField.

### PR DESCRIPTION
**Pull request by doniyor2109**

### CODE REVIEW

File: packages/forms/src/FormikCheckboxField.tsx [hunk 1]
```diff
@@ -41,7 +46,11 @@ export function FormikCheckboxField({
       }}
       onChange={(event, checked) => {
         onChange?.(event, checked);
-        field.onChange(event);
+        if (parse) {
+          setValue(parse(event as ChangeEvent<HTMLInputElement>));
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:50]: well, that's not right too. we should respect upstream types.

Comment (doniyor2109) [reply packages/forms/src/FormikCheckboxField.tsx:50]: Material marked the event as `ChangeEvent<{}>`

File: packages/forms/src/FormikCheckboxField.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import { FieldValidator, useField, useFormikContext } from 'formik';
-import React, { ChangeEvent } from 'react';
+import React from 'react';
 
 export interface FormikCheckboxFieldProps extends CheckboxFieldProps {
   name: string;
   validate?: FieldValidator;
   format?: (value?: boolean) => boolean | undefined;
-  parse?: (event: ChangeEvent<HTMLInputElement>) => boolean;
+  parse?: (event: React.ChangeEvent<Record<string, unknown>>) => boolean;
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:9]:
```suggestion
  parse?: CheckboxFieldProps['onChange']
```

File: packages/forms/src/__tests__/FormikCheckboxField.spec.tsx [hunk 3]
```diff
@@ -41,6 +42,40 @@ test('handles changes', async () => {
   expect(handleSubmit).toHaveBeenLastCalledWith({ agree: false });
 });
 
+test('format and parse value', async () => {
+  const handleSubmit = jest.fn();
+
+  const wrapper = renderFormField(
```
Comment (umidbekkarimov) [inline packages/forms/src/__tests__/FormikCheckboxField.spec.tsx:48]: make a better example with `active`/`inactive` enumerator values. you will find problems in type definitions.

File: packages/forms/src/FormikCheckboxField.tsx [hunk 4]
```diff
@@ -5,12 +5,16 @@ import React from 'react';
 export interface FormikCheckboxFieldProps extends CheckboxFieldProps {
   name: string;
   validate?: FieldValidator;
+  format?: (value?: boolean) => boolean | undefined;
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:8]: `format?: (value: unknown) => boolean | undefined;`

File: packages/forms/src/FormikCheckboxField.tsx [hunk 5]
```diff
@@ -20,7 +24,7 @@ export function FormikCheckboxField({
 }: FormikCheckboxFieldProps) {
   const uid = useUID();
   const { isSubmitting } = useFormikContext();
-  const [field, { error, touched }] = useField({
+  const [field, { error, touched }, { setValue }] = useField<boolean>({
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:27]:
```suggestion
  const [field, { error, touched }, { setValue }] = useField<unknown>({
```

File: packages/forms/src/FormikCheckboxField.tsx [hunk 6]
```diff
@@ -5,12 +5,16 @@ import React from 'react';
 export interface FormikCheckboxFieldProps extends CheckboxFieldProps {
   name: string;
   validate?: FieldValidator;
+  format?: (value?: boolean) => boolean | undefined;
+  parse?: (event: React.ChangeEvent<Record<string, unknown>>) => boolean;
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:9]:
```suggestion
  parse?: (event: React.ChangeEvent<Record<string, unknown>>, checked: boolean) => unknown;
```

File: packages/forms/src/__tests__/FormikCheckboxField.spec.tsx [hunk 7]
```diff
@@ -41,6 +42,42 @@ test('handles changes', async () => {
...
+
+  expect(field).not.toBeChecked();
+
+  wrapper.submitForm();
+
+  await waitFor(() => {
+    expect(handleSubmit).toHaveBeenCalledTimes(1);
+  });
+
+  expect(handleSubmit).toHaveBeenLastCalledWith({ status: 'inactive' });
```
Comment (umidbekkarimov) [inline packages/forms/src/__tests__/FormikCheckboxField.spec.tsx:78]: can you check also for the `false` value result?

Comment (doniyor2109) [reply packages/forms/src/__tests__/FormikCheckboxField.spec.tsx:78]: You mean with for `parse/format`?

Comment (umidbekkarimov) [reply packages/forms/src/__tests__/FormikCheckboxField.spec.tsx:114]: I meant `true`, just click to the checkbox one more time and check submission result

File: packages/forms/src/FormikCheckboxField.tsx [hunk 8]
```diff
@@ -5,12 +5,22 @@ import React from 'react';
 export interface FormikCheckboxFieldProps extends CheckboxFieldProps {
   name: string;
   validate?: FieldValidator;
+  format?: (
+    value: unknown,
+    checked: boolean | undefined,
+  ) => boolean | undefined;
+  parse?: (
+    event: React.ChangeEvent<Record<string, unknown>>,
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:13]:
```suggestion
    event: React.ChangeEvent<HTMLInputElement >,
```

File: packages/forms/src/FormikCheckboxField.tsx [hunk 9]
```diff
@@ -5,12 +5,22 @@ import React from 'react';
 export interface FormikCheckboxFieldProps extends CheckboxFieldProps {
   name: string;
   validate?: FieldValidator;
+  format?: (
+    value: unknown,
+    checked: boolean | undefined,
+  ) => boolean | undefined;
+  parse?: (
+    event: React.ChangeEvent<HTMLInputElement >,
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikCheckboxField.tsx:13]:
```suggestion
    event: React.ChangeEvent<HTMLInputElement>,
```

**Pull request merged**